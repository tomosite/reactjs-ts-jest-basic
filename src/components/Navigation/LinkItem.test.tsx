import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { describe } from 'node:test';
import {LinkItem} from '.';

describe('App', () => {
  test('click the navigation link item', () => {
    const mockFn = jest.fn();
    render(<LinkItem onClick={mockFn}>item dummy 1</LinkItem>);
    const li_elem = screen.getByText('item dummy 1');
    fireEvent.click(li_elem, { button: 0 });
    
    /* item dummy 1 - snapshot */
    expect(li_elem).toMatchSnapshot();

    /* item dummy 1 - is clicked */
    expect(mockFn).toHaveBeenCalled();

    /* item dummy 1 - is found */
    expect(li_elem).toBeTruthy();
  });
});
