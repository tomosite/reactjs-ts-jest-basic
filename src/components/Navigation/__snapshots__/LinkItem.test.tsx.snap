// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click the navigation link item 1`] = `
<li>
  item dummy 1
</li>
`;
